src/java/map/BinaryMask.java

```diff
@@ -134,6 +134,39 @@ public boolean isEdge(int x, int y) {
                 || (y < getSize() - 1 && get(x, y + 1) != value));
     }
 
+    public BinaryMask setRectangularAreaWithoutSymmetry(int x1, int x2, int z1, int z2, boolean trueOrFalse) {
```

Comment:
this function already exists. See fillRect and for more arbitrary see FillParallelogram